src/internal/single-tab-stop/keyboard-nav-util.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as React from 'react';
+import { SingleTabStopNavigationAPI, SingleTabStopNavigationProvider } from '../../../src/internal/single-tab-stop';
+
+import handleKey from '../../../src/internal/utils/handle-key';
+import { circleIndex } from '../../../src/internal/utils/circle-index';
+import { KeyCode } from '../../../src/internal/keycode';
+import { getAllFocusables } from '../../../src/internal/focus-lock/utils';
+
+export default function SingleTabStopKeyboardNav({
+  children,
+  matchId,
```

Comment:
the matchId name is not very telling - it is actually a selector that finds interactive elements that can be focus targets